### PR TITLE
planNotification

### PR DESCRIPTION
**Pull request by OfirZinkerz**

### CODE REVIEW

File: src/components/planNotification/run/planNotification.run.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+(function(){
+    'use strict';
+
+    angular.module('znk.infra-web-app.planNotification').run(
+        function checkPlanNotification(NotificationService, NotificationTypeEnum, ENV, ZnkServiceEnum,
+                                       PlanNotificationService, PlanService, $log, AuthService, $location){
```
Comment (choroshin) [inline src/components/planNotification/run/planNotification.run.js]: replace PlanService with API request

File: src/components/planNotification/run/planNotification.run.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+                    $log.debug('checkPlanNotification: connectStudentToPlan successful');
+                    delete search.planId;
+                    $location.search(search);
+                }, function errorCallback(err) {
+                    $log.error('checkPlanNotification: error in PlanService.connectStudentToPlan, err: ' + err);
+                });
+            }
+        }
+    );
+})();
```
Comment (choroshin) [inline src/components/planNotification/run/planNotification.run.js]: add (angular);

File: src/components/planNotification/run/planNotification.run.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+            var search = $location.search();
+            if (angular.isDefined(search.planId)) {
+                var uid = AuthService.getAuth().uid;
+                var connectStudentToPlanUrl = ENV.myZinkerz + '/plan/connectStudentToPlan';
+                $http({
+                    method: 'POST',
+                    url: connectStudentToPlanUrl,
+                    data: { planId: search.planId, uid: uid }
+                }).then(function successCallback() {
+                    PlanNotificationService.newPlanNotification({ refObjId: search.planId });
```
Comment (Ariel-W) [inline src/components/planNotification/run/planNotification.run.js]: Do not show the popup here, you will get an event from NotificationService.

File: src/components/planNotification/run/planNotification.run.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
+(function(){
+    'use strict';
+
+    angular.module('znk.infra-web-app.planNotification').run(
+        function checkPlanNotification(NotificationService, NotificationTypeEnum, ZnkServiceEnum, $http,
+                                       PlanNotificationService, $log, AuthService, $location, ENV){
+            'ngInject';
+            NotificationService.clean(NotificationTypeEnum.planCreated);
+            NotificationService.on(NotificationTypeEnum.planCreated, PlanNotificationService.newPlanNotification);
```
Comment (Ariel-W) [inline src/components/planNotification/run/planNotification.run.js]: planPending

**Comment by OfirZinkerz** [comment]
jenkins-test

**Comment by OfirZinkerz** [comment]
jenkins-test

**Comment by OfirZinkerz** [comment]
jenkins-test

**Comment by OfirZinkerz** [comment]
jenkins-test

**Comment by assafshp** [comment]
jenkins-test

**Comment by assafshp** [comment]
jenkins-test

**Comment by assafshp** [comment]
jenkins-test

**Comment by assafshp** [comment]
jenkins-test

File: src/components/planNotification/services/planNotification.service.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+            function _getPlanIdFromUrl() {
+                var search = $location.search();
+                return angular.isDefined(search.planId) ? search.planId : null;
+            }
+
+            function _checkPlanNotification(){
+                var planId = _getPlanIdFromUrl();
+                if (planId) {
+                    var uid = AuthService.getAuth().uid;
+                    var connectStudentToPlanUrl = ENV.myZinkerz + '/plan/connectStudentToPlan';
```
Comment (Ariel-W) [inline src/components/planNotification/services/planNotification.service.js]: ENV.myZinkerz => ENV.zinkerzBE

File: src/components/planNotification/services/planNotification.service.js [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+                        $log.debug('checkPlanNotification: connectStudentToPlan successful');
+                    }, function errorCallback(err) {
+                        $log.error('checkPlanNotification: error in PlanService.connectStudentToPlan, err: ' + err);
+                    });
+                }
+            }
+
+            function _newPlanNotification(notification) {
+                _showPlanNotificationPopUp().then(function () {
+                    $window.open(ENV.myZinkerz + '?planId=' + notification.refObjId);
```
Comment (Ariel-W) [inline src/components/planNotification/services/planNotification.service.js]: Route directly to: ENV.myZinkerz + '/newPlan"

Discuss with Assaf regarding if we can have the "newPlan" component without a template, only logic...

File: src/components/planNotification/services/planNotification.service.js (hunk 5 above)
Comment (OfirZinkerz) [inline src/components/planNotification/services/planNotification.service.js]: fixed

**Pull request merged**